### PR TITLE
Fix conversion of keyframe timestamps from child to parent animations

### LINKED ISSUE #61: Child animation keyframes are not guaranteed to render when parent animation has a non-linear curve
The `Animation` API is designed to guarantee that all keyframes will be rendered. This ensures that all keyframe-animated properties hit their start and end values. This applies to child animations as well, but there is a bug in the way we translate the relative timestamps that causes this to only work when the parent animation has a linear curve.

See the following example:

```swift
var labelAnimation = Animation<UILabel>()
labelAnimation.addKeyframe(for: \.alpha, at: 0, value: 1)
labelAnimation.addKeyframe(for: \.alpha, at: 1, value: 0)

labelAnimation.addPerFrameExecution { context in
    print("Setting label alpha to \(context.element.alpha)")
}

var animationGroup = AnimationGroup()
animationGroup.curve = CubicBezierAnimationCurve.easeInEaseOut
animationGroup.addAnimation(labelAnimation, for: label, startingAt: 0, relativeDuration: 0.2)
```

```
Setting label alpha to 1.0
Setting label alpha to 0.9952884912490845
Setting label alpha to 0.9809672832489014
Setting label alpha to 0.9567672610282898
Setting label alpha to 0.9224337339401245
Setting label alpha to 0.8777324557304382
Setting label alpha to 0.8224567174911499
Setting label alpha to 0.756435751914978
Setting label alpha to 0.6795437932014465
Setting label alpha to 0.5917111039161682
Setting label alpha to 0.4929352104663849
Setting label alpha to 0.3832937777042389
Setting label alpha to 0.26295730471611023
Setting label alpha to 0.13220225274562836
```

As you can see, the last keyframe of the `labelAnimation` is never rendered, leaving the label at an alpha greater than zero.

### PR DESCRIPTION
**Pull request by NickEntin**
Fixes #61

### CODE REVIEW

File: Sources/Stagehand/Animation/Animation.swift [hunk 1]
```diff
@@ -93,14 +93,23 @@ public struct Animation<ElementType: AnyObject> {
...
-            }
+
+            let adjustedRelativeTimestamps: [Double] = childKeyframeRelativeTimestamps
+                .flatMap { childRelativeTimestamp -> [Double] in
+                    /// Convert the timestamp to be relative to the curved progress of this animation by adjusting the
+                    /// range to the start timestamp and duration of the child.
+                    let timestampInParent = child.relativeStartTimestamp + childRelativeTimestamp * child.relativeDuration
+
+                    /// Convert the timestamp to be relative to the uncurved progress of this animation.
+                    return curve.rawProgress(for: timestampInParent)
```
Comment (NickEntin) [inline Sources/Stagehand/Animation/Animation.swift:111]: This is the crux of the fix.

File: Sources/Stagehand/AnimationCurve/AnimationCurve.swift [hunk 2]
```diff
@@ -27,4 +27,6 @@ public protocol AnimationCurve {
 
     func adjustedProgress(for progress: Double) -> Double
 
+    func rawProgress(for adjustedProgress: Double) -> [Double]
```
Comment (NickEntin) [inline Sources/Stagehand/AnimationCurve/AnimationCurve.swift:34]: Probably worth a comment explaining why `adjustedProgress(for:)` returns a `Double`, but `rawProgress(for:)` returns a `[Double]`.

```suggestion
    /// The raw (uncurved) progress values that correspond to the specified adjusted (curved) progress.
    ///
    /// Note that unlike the raw -> adjusted calculation that always results in one value, there may be multiple raw
    /// values corresponding to one adjusted value. In other words, X(t) is monotonic, but Y(t) is not.
    func rawProgress(for adjustedProgress: Double) -> [Double]
```

Comment (NickEntin) [inline Sources/Stagehand/AnimationCurve/AnimationCurve.swift:34]: Probably worth a comment explaining why `adjustedProgress(for:)` returns a `Double`, but `rawProgress(for:)` returns a `[Double]`.

```suggestion
    /// The raw (uncurved) progress values that correspond to the specified adjusted (curved) progress.
    ///
    /// Note that unlike the raw -> adjusted calculation that always results in one value, there may be multiple raw
    /// values corresponding to one adjusted value. In other words, X(t) is monotonic, but Y(t) is not.
    func rawProgress(for adjustedProgress: Double) -> [Double]
```

**Review by cocoalabs — APPROVED** [review]
Lgtm

**Pull request merged**